src/UserContext.js

```diff
@@ -41,10 +41,17 @@ function userReducer(state, action) {
         case UserActionTypes.Update:
             return { ...state, ...action.payload };
         case UserActionTypes.Register:
-            return {
-                isLoggedInUser: true, email: action.payload.email,
-                name: action.payload.name
-            };
+
+            if (
```

Comment:
Any user should be able to register. Don't check payload for a specific user for the Register action. Instead, you could have a variable (e.g. array) where you push a new object with user information when user registers. 
It's a very basic solution that will only work until page is reloaded but is good enough for now.
E.g. define variable in this file but outside the reducer functions: `let registeredUsers = [{name: 'johndoe', email: 'johndoe@gmail.com', password: '12345'}]`. Then, you can push to this array new users when Register operation happens.